inc/ChangeLog/RevisionInfo.php

```diff
@@ -115,146 +166,167 @@ public function editor()
      *
      * @return string
      */
-    public function itemName()
+    public function showFileName()
     {
-        $id = $this->info['id'];
-        $rev = ($this->info['current']) ? '' : $this->info['date'];
+        $id = $this->val('id');
+        $rev = $this->isCurrent() ? '' : $this->val('date');
 
-        switch ($this->info['item']) {
+        switch ($this->val('mode')) {
             case 'media': // media file revision
                 $params = ['tab_details'=> 'view', 'ns'=> getNS($id), 'image'=> $id];
                 if ($rev) $params += ['rev'=> $rev];
                 $href = media_managerURL($params, '&');
-                if(file_exists(mediaFN($id, $rev))) {
-                    $class = 'wikilink1';
-                } else {
-                    $class = 'wikilink2';
-                    if(!$this->info['current']) {
-                        //revision is not in attic
-                        return $id;
-                    }
-                }
-                return '<a href="'.$href.'" class="'.$class.'">'.$id.'</a>';
+                $display_name = $id;
+                $class = file_exists(mediaFN($id, $rev)) ? 'wikilink1' : 'wikilink2';
+                break;
             case 'page': // page revision
                 $params = $rev ? ['rev'=> $rev] : [];
                 $href = wl($id, $params, false, '&');
                 $display_name = useHeading('navigation') ? hsc(p_get_first_heading($id)) : $id;
                 if (!$display_name) $display_name = $id;
-                if(page_exists($id, $rev)) {
-                    $class = 'wikilink1';
-                } else {
-                    $class = 'wikilink2';
-                    if(!$this->info['current']) {
-                        //revision is not in attic
-                        return $display_name;
-                    }
-                }
-                return '<a href="'.$href.'" class="'.$class.'">'.$display_name.'</a>';
+                $class = page_exists($id, $rev) ? 'wikilink1' : 'wikilink2';
+        }
+        if ($this->val('type') == DOKU_CHANGE_TYPE_DELETE) {
+            $class = 'wikilink2';
+        }
+        return '<a href="'.$href.'" class="'.$class.'">'.$display_name.'</a>';
```

Comment:
See also 3d727ea43c26288d1999e69448e2651bdc41e32e

Three ways of formatting of the url are needed:
- internal link green, if exists. If you follow you arrive at a existing page (current or from attic)
- internal link red, if does not exist. If you follow you arrive at not-existing page (useful to directly recreate a page from scratch/upload new media, so not recover old)
- just plain text, if the page does not exist in attic. There is nothing to follow.

Used in the following manners
Revisions page of a wiki page/media:
- first line current revision: green of red link, if followed you get last page or you can directly create/upload a (completely) new version.
- all other lines: green link or plain text, older revisions, all from attic. If followed you see old page/media from attic. If you follow you can restore with the restore buttton. If item does not exist in attic, it should not be a link because there is nothing there.

Recent changes page:
- lists of only current wiki pages/media (so only layouting of first line of Revisions page applies):
 green of red link, if followed you get last page or you can directly create/upload a (completely) new version.

In my understanding this is not dependent on DOKU_CHANGE_TYPE_DELETE, because green/red should in Revisions page predict if you can follow the link, and are already determined by page_exists() for current page. For the other lines of the Revisions page, the link should appear as plain text if the revision does not exists in the attic and cannot be followed.

Reply:
I see DOKU_CHANGE_TYPE_DELETE is different question than linking to attic. See https://github.com/splitbrain/dokuwiki/pull/3607#issuecomment-1019573657